### ISSUE TITLE
scratch.MapSet  can't set a array value

### ISSUE THREAD

**Issue by movekj** [hashicorp/consul-template#1097]
this does not report error: 
```bash 
$ cat test.tmp
{{ scratch.MapSet "tags" "v1" 1 }}
{{ scratch.MapSet "tags" "v2" 1 }}
{{ scratch.MapSet "tags" "v3" 1 }}

$ consul-template -template "~/test.tmp:/tmp/test.txt" -consul-addr=127.0.0.1:8501 -once
```
but: 
```bash
$ cat test.ctmp
{{ scratch.MapSet "obj" "tags" ["v1", "v2", "v3"] }}

$ consul-template -template "~/test.ctmp:/tmp/test.txt" -consul-addr=127.0.0.1:8501 -once
2018/04/25 02:10:46.108056 [ERR] (cli) /home/lixingxing/test.ctmp: parse: template: :1: unexpected "[" in operand
```
**can i set  array to a map key use scratch.MapSet?**

**Comment by eikenb** [comment]
Hey @movekj, sorry for the long delay in a response.

The template syntax uses the standard Go text/template syntax and in this case the important bit is on [Arguments](https://golang.org/pkg/text/template/#hdr-Arguments). 

TLDR; slice/array syntax isn't supported for template arguments.

In the future, for questions such as this, may I refer you to our community support (below) as this tracker is meant for bugs. Thanks.

https://github.com/hashicorp/consul-template#community-support

https://discuss.hashicorp.com/c/consul
https://www.consul.io/community.html
